app/controllers/concerns/saml_idp_auth_concern.rb

```diff
@@ -148,12 +148,22 @@ def saml_response
       name_id_format: name_id_format,
       authn_context_classref: requested_authn_context,
       reference_id: active_identity.session_uuid,
-      encryption: current_service_provider.encryption_opts,
+      encryption: encryption_opts,
       signature: saml_response_signature_options,
       signed_response_message: current_service_provider.signed_response_message_requested,
     )
   end
 
+  def encryption_opts
+    url = URI.parse request.original_url
+    query_params = Rack::Utils.parse_nested_query url.query
+    if query_params['skip_encryption'].present? && current_service_provider.skip_encryption_allowed
```

Comment:
We have `UriService.params` if we want to save a few lines:

```suggestion
    query_params = UriService.params(request.original_url)
    if query_params[:skip_encryption].present? && current_service_provider.skip_encryption_allowed
```